packages/txservice/src/dispatch.ts

```diff
@@ -516,11 +548,32 @@ export class TransactionDispatch extends ChainRpcProvider {
       });
     }
 
-    // Get receipt for tx with at least 1 confirmation. If it times out (using default, configured timeout),
-    // it will throw a TransactionTimeout error.
-    const result = await this.confirmTransaction(transaction, 1);
-    if (result.isErr()) {
-      const { error: _error } = result;
+    try {
+      // Get receipt for tx with at least 1 confirmation. If it times out (using default, configured timeout),
+      // it will throw a TransactionTimeout error.
+      const receipt = await this.confirmTransaction(transaction, 1);
+
+      // Sanity checks.
+      if (receipt.status === 0) {
+        // This should never occur. We should always get a TransactionReverted error in this event.
+        throw new TransactionProcessingError(TransactionProcessingError.reasons.DidNotThrowRevert, method, {
+          chainId: this.chainId,
+          receipt,
+          transaction: transaction.loggable,
+        });
```

Comment:
same as above -- if there's anything else you think should be logged here include it in the error context